### PR TITLE
Add support to parallel pregen to only generate angle signatures

### PR DESCRIPTION
**Pull request by walksanatora**
adds the `-o --only-tail` argument which generates only the tail

### CODE REVIEW

**Comment by walksanatora** [comment]
to make you feel save (so git hash matches) 
image

File: src/bin/pregen.rs [hunk 1]
```diff
@@ -81,6 +81,14 @@ fn main() {
         }
         None => find_patterns(all_targets),
     };
-
-    fs::write(format!("numbers_{}.json", cli.max), serde_json::to_string_pretty(&all_data).unwrap()).unwrap();
+    if cli.only_tail {
+        fs::write(format!("numbers_{}.json", cli.max), serde_json::to_string_pretty(&all_data).unwrap()).unwrap();
+    } else {
+        let mut pos_neg_values: BTreeMap<i64, (String, String)> = BTreeMap::new();
+        for (k, v) in all_data {
```
Comment (object-Object) [inline src/bin/pregen.rs]: Missing the zero check, does the resulting file have an entry for -0? Also would prefer more descriptive variable names here

File: src/bin/pregen.rs [hunk 2]
```diff
@@ -81,6 +81,14 @@ fn main() {
...
         None => find_patterns(all_targets),
     };
-
-    fs::write(format!("numbers_{}.json", cli.max), serde_json::to_string_pretty(&all_data).unwrap()).unwrap();
+    if cli.only_tail {
+        fs::write(format!("numbers_{}.json", cli.max), serde_json::to_string_pretty(&all_data).unwrap()).unwrap();
+    } else {
+        let mut pos_neg_values: BTreeMap<i64, (String, String)> = BTreeMap::new();
+        for (k, v) in all_data {
+            pos_neg_values.insert(k, (Direction::SouthEast.to_string(), "aqaa".to_owned() + &v));
```
Comment (object-Object) [inline src/bin/pregen.rs]: Would prefer `format!("aqaa{v}")`

File: src/bin/pregen.rs [hunk 3]
```diff
@@ -81,6 +81,14 @@ fn main() {
         }
         None => find_patterns(all_targets),
     };
-
-    fs::write(format!("numbers_{}.json", cli.max), serde_json::to_string_pretty(&all_data).unwrap()).unwrap();
+    if cli.only_tail {
+        fs::write(format!("numbers_{}.json", cli.max), serde_json::to_string_pretty(&all_data).unwrap()).unwrap();
```
Comment (object-Object) [inline src/bin/pregen.rs]: Can you refactor this line out into a generic function?

File: src/bin/pregen.rs [hunk 4]
```diff
@@ -22,37 +22,37 @@ fn n_groups<T>(mut values: Vec<T>, n: usize) -> Vec<Vec<T>> {
...
     tx.send(find_patterns(targets)).unwrap();
 }
 
 #[derive(Parser)]
+#[command(author = "Gamma Delta", version = "1", about = "generates hexcasting number literals")]
 struct Cli {
     /// Largest number to generate a literal for.
     max: u64,
 
+    /// only generate the "tail" of the number literal
```
Comment (object-Object) [inline src/bin/pregen.rs]:
```suggestion
    /// Only generate the "tail" of the number literal (skip aqaa/dedd).
```

File: src/bin/pregen.rs [hunk 5]
```diff
@@ -22,37 +22,37 @@ fn n_groups<T>(mut values: Vec<T>, n: usize) -> Vec<Vec<T>> {
...
     data
 }
 
-fn worker(targets: Vec<i64>, tx: Sender<BTreeMap<i64, (String, String)>>) {
+fn worker(targets: Vec<i64>, tx: Sender<BTreeMap<i64, String>>) {
     tx.send(find_patterns(targets)).unwrap();
 }
 
 #[derive(Parser)]
+#[command(author = "Gamma Delta", version = "1", about = "generates hexcasting number literals")]
```
Comment (object-Object) [inline src/bin/pregen.rs]: Would prefer using the syntax to automatically pull these from Cargo.toml. Though honestly I don't think this line is really necessary at all.

**Review by object-Object — CHANGES_REQUESTED** [review]

File: src/bin/pregen.rs (hunk 5 above)
Comment (walksanatora) [reply src/bin/pregen.rs]: does clap have that option?

File: src/bin/pregen.rs (hunk 1 above)
Comment (walksanatora) [reply src/bin/pregen.rs]: gonna test if rust is smart enough to -0==0

Comment (walksanatora) [reply src/bin/pregen.rs]: yeah they are equivalent

Comment (walksanatora) [reply src/bin/pregen.rs]: but it is a tiny bit of wasted processing power

File: src/bin/pregen.rs (hunk 2 above)
Comment (walksanatora) [reply src/bin/pregen.rs]: forgot format existed

**Comment by walksanatora** [comment]
that should be it?

File: src/bin/pregen.rs [hunk 6]
```diff
@@ -82,5 +81,21 @@ fn main() {
         None => find_patterns(all_targets),
     };
 
-    fs::write(format!("numbers_{}.json", cli.max), serde_json::to_string_pretty(&all_data).unwrap()).unwrap();
+    fs::write(
+        format!("numers_{}.json", cli.max),
```
Comment (object-Object) [inline src/bin/pregen.rs]:
```suggestion
        format!("numbers_{}.json", cli.max),
```

File: src/bin/pregen.rs (hunk 5 above)
Comment (object-Object) [reply src/bin/pregen.rs]: yes: https://docs.rs/clap/latest/clap/_derive/_tutorial/index.html#configuring-the-parser

File: src/bin/pregen.rs [hunk 7]
```diff
@@ -82,5 +81,21 @@ fn main() {
...
 
-    fs::write(format!("numbers_{}.json", cli.max), serde_json::to_string_pretty(&all_data).unwrap()).unwrap();
+    fs::write(
+        format!("numers_{}.json", cli.max),
+        if cli.only_tail {
+            serde_json::to_string_pretty(&all_data).unwrap()
+        } else {
+            let mut pos_neg_values: BTreeMap<i64, (String, String)> = BTreeMap::new();
+            for (k, v) in all_data {
+                if k == 0 {
```
Comment (object-Object) [inline src/bin/pregen.rs]: Now it just doesn't insert an entry for zero at all.

**Review by object-Object — CHANGES_REQUESTED** [review]

**Comment by walksanatora** [comment]
so i see you cleaned up the code a bit and added a --pretify argument
so does this mean LGTM?

**Review by object-Object — APPROVED** [review]

**Comment by object-Object** [comment]
Thanks for the PR!

**Pull request merged**